web/client-api/src/main/java/io/deephaven/web/client/api/subscription/TableSubscription.java

```diff
@@ -31,80 +24,41 @@
  * viewports to make it less expensive to compute for large tables.
  */
 @JsType(namespace = "dh")
-public class TableSubscription extends HasEventHandling {
-
-    /**
-     * Indicates that some new data is available on the client, either an initial snapshot or a delta update. The
-     * <b>detail</b> field of the event will contain a TableSubscriptionEventData detailing what has changed, or
-     * allowing access to the entire range of items currently in the subscribed columns.
-     */
-    public static final String EVENT_UPDATED = "updated";
-
+public final class TableSubscription extends AbstractTableSubscription {
 
-    // column defs in this subscription
-    private JsArray<Column> columns;
-    // holder for data
-    private SubscriptionTableData data;
+    private final JsArray<Column> columns;
+    private final Double updateIntervalMs;
 
-    // table created for this subscription
-    private Promise<JsTable> copy;
-
-    // copy from the initially given table so we don't need to way
     @JsIgnore
     public TableSubscription(JsArray<Column> columns, JsTable existingTable, Double updateIntervalMs) {
-
-        copy = existingTable.copy(false).then(table -> new Promise<>((resolve, reject) -> {
-            table.state().onRunning(newState -> {
-                // TODO handle updateInterval core#188
-                table.internalSubscribe(columns, this);
-
-                resolve.onInvoke(table);
-            }, table::close);
-        }));
-
+        super(existingTable.state(), existingTable.getConnection());
         this.columns = columns;
-        Integer rowStyleColumn = existingTable.state().getRowFormatColumn() == null ? NO_ROW_FORMAT_COLUMN
-                : existingTable.state().getRowFormatColumn().getIndex();
-        this.data = new SubscriptionTableData(columns, rowStyleColumn, this);
-
+        this.updateIntervalMs = updateIntervalMs;
     }
 
-    // public void changeSubscription(JsArray<Column> columns) {
-    // copy.then(t ->{
-    // t.internalSubscribe(columns, this);
-    // return Promise.resolve(t);
-    // });
-    // this.columns = columns;
-    // }
-
-
-    @JsIgnore
-    public void handleSnapshot(TableSnapshot snapshot) {
-        data.handleSnapshot(snapshot);
-    }
-
-    @JsIgnore
-    public void handleDelta(DeltaUpdates delta) {
-        data.handleDelta(delta);
+    @Override
+    protected void sendFirstSubscriptionRequest() {
+        changeSubscription(columns, updateIntervalMs);
     }
 
     /**
-     * The columns that were subscribed to when this subscription was created
+     * Updates the subscription to use the given columns and update interval.
```

Comment:
As discussed on slack, interval cannot change once the initial subscription has been sent. This is a technical limitation where we are trying to share the BarrageMessageProducer with as many subscribers as possible -- but they must be using the same update interval to share the BMP. 

The BMP is only constructed when trying to resolve the initial table to subscribe to.

Reply:
Okay - viewport already errors if this is changed, will change non-viewport subs to do the same. I'll make a note to track this, but we could change this later to re-create the subscription if that value changes.